lib/cassandra/protocol/requests/query_request.rb

```diff
@@ -66,7 +66,11 @@ def write(buffer, protocol_version, encoder)
             buffer.append(flags.chr)
```

Comment:
Shouldn't we test `if protocol_version < 5` here as well?

Reply:
There's probably an argument that could be made along the lines of "for this ticket we only need to fix things enough to allow connection negotiation to complete and we can do that without the other case"... but that seems kinda pedantic to me.  Fixing since we're already here, good catch.